# HG changeset patch
# User Rik <rik@octave.org>
# Date 1451504609 28800
#      Wed Dec 30 11:43:29 2015 -0800
# Node ID c22219b8a6a6775e7a04eccc0cb8494dc0bfac5c
# Parent  b009c65ec062634c12c0827e2f22e9319e0ab4c9
Silence 'no return' warnings during compilation of liboctave

* lo-error.h (*liboctave_error_handler, *liboctave_error_with_id_handler):
Mark functions with GCC_ATTR_NORETURN.

diff --git a/liboctave/cruft/misc/lo-error.h b/liboctave/cruft/misc/lo-error.h
--- a/liboctave/cruft/misc/lo-error.h
+++ b/liboctave/cruft/misc/lo-error.h
@@ -30,19 +30,19 @@ extern "C" {
 extern void liboctave_fatal (const char *fmt, ...) GCC_ATTR_NORETURN;
 
 extern void liboctave_fatal_with_id (const char *id, const char *fmt, ...) GCC_ATTR_NORETURN;
 
 extern void liboctave_warning (const char *fmt, ...);
 
 extern void liboctave_warning_with_id (const char *id, const char *fmt, ...);
 
-typedef void (*liboctave_error_handler) (const char *, ...);
+typedef void (*liboctave_error_handler) (const char *, ...) GCC_ATTR_NORETURN;
 
-typedef void (*liboctave_error_with_id_handler) (const char *, const char *, ...);
+typedef void (*liboctave_error_with_id_handler) (const char *, const char *, ...) GCC_ATTR_NORETURN;
 
 typedef void (*liboctave_warning_handler) (const char *, ...);
 
 typedef void (*liboctave_warning_with_id_handler) (const char *, const char *, ...);
 
 /* Would be nice to make these pointers private, but we want to share
    them among all the liboctave classes. */
 CRUFT_API extern liboctave_error_handler current_liboctave_error_handler;
